From cf06dcb68c251896dba1777984c93e791a47ac89 Mon Sep 17 00:00:00 2001
From: Christian Fetzer <fetzer.ch@gmail.com>
Date: Sun, 4 Dec 2016 14:23:53 +0100
Subject: [PATCH 1/1] Fix compilation with DEBUG=1

The core fails to build with DEBUG=1 because it relies on symbols being
optimized out. Therefore switch from O0 to Og (optimize for debugging).

libretro.o: In function `port_rbyte(int&, unsigned int)':
libretro.cpp:(.text+0x784): undefined reference to `HuC6273_Read8(unsigned int)'
libretro.cpp:(.text+0x7e4): undefined reference to `FXSCSI_CtrlRead(unsigned int)'
libretro.o: In function `port_rhword(int&, unsigned int)':
libretro.cpp:(.text+0xa1f): undefined reference to `HuC6273_Read16(unsigned int)'
libretro.cpp:(.text+0xa8b): undefined reference to `FXSCSI_CtrlRead(unsigned int)'
libretro.o: In function `port_wbyte(int&, unsigned int, unsigned char)':
libretro.cpp:(.text+0xcfe): undefined reference to `HuC6273_Write16(unsigned int, unsigned short)'
libretro.cpp:(.text+0xd2e): undefined reference to `FXSCSI_CtrlWrite(unsigned int, unsigned char)'
(...)
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 71b79b8..83ddcf7 100644
--- a/Makefile
+++ b/Makefile
@@ -256,7 +256,7 @@ all: $(TARGET)
 ifeq ($(DEBUG),0)
    FLAGS += -O2 $(EXTRA_GCC_FLAGS)
 else
-   FLAGS += -O0
+   FLAGS += -Og
 endif
 
 LDFLAGS += $(fpic) $(SHARED)
-- 
2.11.0

